### PR TITLE
Added a queue data structure.

### PR DESCRIPTION
**Pull request by tgrondier**

### CODE REVIEW

File: src/Queue.cpp [hunk 1]
```diff
@@ -1,6 +1,53 @@
...
 
-Queue::Queue()
+
+namespace QMLExtraDataStructures {
+Queue::Queue(QQuickItem *parent) : QQuickItem(parent) {
+    queue = QQueue<QVariant>();
+}
+
+void Queue::clear() {
+    queue = QQueue<QVariant>();
```
Comment (ayberkozgur) [inline src/Queue.cpp]: Why not `.clear()` (inherited from `QList`)?

**Review by ayberkozgur — CHANGES_REQUESTED** [review]
`clear()`

**Review by ayberkozgur — CHANGES_REQUESTED** [review]
`clear()`

File: src/Queue.cpp [hunk 2]
```diff
@@ -1,6 +1,53 @@
...
+ */
+
+
 #include "Queue.h"
 
-Queue::Queue()
+
+namespace QMLExtraDataStructures {
+Queue::Queue(QQuickItem *parent) : QQuickItem(parent) {
+    queue = QQueue<QVariant>();
```
Comment (ayberkozgur) [inline src/Queue.cpp]: Not necessary

**Review by ayberkozgur — CHANGES_REQUESTED** [review]
check `clear()` and constructor

File: src/Queue.cpp (hunk 2 above)
Comment (tgrondier) [reply src/Queue.cpp]: I don't understand what you mean by not necessary. Could you please be more precise ?

File: src/Queue.cpp (hunk 1 above)
Comment (tgrondier) [reply src/Queue.cpp]: I missed the QList inheritance while writing the wrapper, fixed.

File: src/Queue.cpp (hunk 2 above)
Comment (ayberkozgur) [reply src/Queue.cpp]: `queue` is already an empty `QQueue<QVariant>` by virtue of the constructor, no need to include the line `queue = QQueue<QVariant>();`

Comment (tgrondier) [reply src/Queue.cpp]: Fixed

**Comment by ayberkozgur** [comment]
Thanks!

**Pull request merged**